work/v1/types.go

```diff
@@ -39,6 +39,11 @@ type ManifestWorkSpec struct {
 	// ManifestConfigs represents the configurations of manifests defined in workload field.
 	// +optional
 	ManifestConfigs []ManifestConfigOption `json:"manifestConfigs,omitempty"`
+
+	// Executor is the configuration of how work agent executes/applies the workloads to the local managed
+	// cluster.
+	// +optional
+	Executor *ManifestWorkExecutor `json:"executor,omitempty"`
```

Comment:
is it because we want backward compatible so we set it as a pointer? I think we should doc the meaning of setting this as nil.

Reply:
added doc about the nil executor